src/targets/gpu/kernels/include/migraphx/kernels/tuple.hpp

```diff
@@ -142,12 +142,12 @@ struct tuple : tuple_detail::tuple_base<Ts...>
     friend constexpr bool operator<(const tuple& x, const tuple& y)
     {
         return x([&](const auto&... xs) {
-            return y([&](const auto&... ys) {
-                fold([&](auto a, auto b) { return a == 0 ? b() : 0; })(0, [&] {
-                    return (xs < ys) ? -1 : (ys < xs) ? 1 : 0;
-                }...);
-            });
-        });
+                   return y([&](const auto&... ys) {
+                       return fold([&](auto a, auto b) { return a == 0 ? b() : a; })(0, [&] {
+                           return (xs < ys) ? -1 : (ys < xs) ? 1 : 0;
+                       }...);
+                   });
+               }) < 0;
```

Comment:
So the fold is returning -1 if its less than, 0 if its equal and 1 if its greater than(ie `(xs < ys) ? -1 : (ys < xs) ? 1 : 0`). As you can see, if the tuple is greater than, it will return true for less than which is the opposite of what you want.